prog/vnet/rekey_nic.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+class Prog::Vnet::RekeyNic < Prog::Base
+  subject_is :nic
+
+  def start
+    nic.dst_ipsec_tunnels.each do |tunnel|
+      args = frame["payload"][tunnel.src_nic.id]
+      create_state(tunnel, args)
+    end
+
+    hop :wait_outbound_trigger
+  end
+
+  def wait_outbound_trigger
+    if SemSnap.new(nic.id).set?("trigger_outbound_update")
+      SemSnap.new(nic.id).decr("trigger_outbound_update")
+      hop :setup_outbound
+    end
+    nap 1
+  end
+
+  def setup_outbound
+    nic.src_ipsec_tunnels.each do |tunnel|
```

Comment:
This traversal is hairy under concurrency much like https://github.com/ubicloud/ubicloud/pull/403/files#r1290882658; need a sketch of a proof about why concurrent modifications are safe, in a note here, or we need a more robust synchronization model. Or we accept it as errata and then pinky swear to do something about it.